lib/build.gradle

```diff
@@ -64,3 +66,14 @@ compileKotlin {
 jar {
     duplicatesStrategy = DuplicatesStrategy.EXCLUDE
 }
+
+tasks.named('compileGroovy') {
+    // Groovy only needs the declared dependencies
+    // (and not longer the output of compileJava)
+    classpath = sourceSets.main.compileClasspath
+}
+tasks.named('compileKotlin') {
+    // Kotlin also depends on the result of Groovy compilation
+    // (which automatically makes it depend of compileGroovy)
+    classpath += files(sourceSets.main.groovy.classesDirectory)
```

Comment:
Wait, why?

Reply:
This :eyes: https://stackoverflow.com/questions/36214437/compile-groovy-and-kotlin

Reply:
If Groovy gets used for tests, we want it to depend on Kotlin outputs, not the other way around.